### PR TITLE
feat: Add new proxy contract

### LINKED ISSUE #73: Proxy contract to relay messages to vaults
@kmchmk you know what to do :wink:

### LINKED ISSUE #74: Vault to whitelist proxy contract and verify sender

### PR DESCRIPTION
**Pull request by kmchmk**
Fix #73 
Fix #74

### CODE REVIEW

**Review by isakfm — APPROVED** [review]
Looks good

**Review by rafaelChelo — CHANGES_REQUESTED** [review]

File: contracts/multisig/src/contract.rs [hunk 1]
```diff
@@ -61,17 +62,62 @@ pub fn execute(
     info: MessageInfo,
     msg: ExecuteMsg,
 ) -> Result<Response<Empty>, ContractError> {
+    // Load the config once
+    let cfg = CONFIG.load(deps.storage)?;
+
+    // Validate the info.sender equal to cfg.proxy if it is set. Then creates a new MessageInfo with sender
+    fn validate_sender(info: &MessageInfo, cfg: &Config, sender: Option<Addr>) -> MessageInfo {
```
[the remark this answers is not in the document]
Comment (kmchmk) [reply contracts/multisig/src/contract.rs:69]: Actually the suggested `from` is the `sender` here.
There are 3 values.
1. cgf.proxy (Allowed proxy address)
2. sender (Sent in every msg. Same as the suggested `from`)
3. info.sender (The actual executor/signer)

The logic here is to validate the `info.sender` is equal to `cfg.proxy`.
Then assign the `sender` to `info.sender`

File: contracts/proxy/src/contract.rs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+#[entry_point]
+pub fn execute(
+    _deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    let cosmsg = CosmosMsg::Wasm(WasmMsg::Execute {
```
[the remark this answers is not in the document]
Comment (kmchmk) [reply contracts/proxy/src/contract.rs:35]: Added the new logic. However the new field is called `sender` instead of `from`.

**Comment by rafaelChelo** [comment]
Well done @kmchmk !
Merging

**Pull request merged**